packages/synapse-react-client/src/components/DirectDownload/DirectDownload.tsx

```diff
@@ -93,36 +87,42 @@ function DirectDownloadIcon(props: DirectDownloadIconProps) {
   return <></>
 }
 
-const DirectDownload: React.FunctionComponent<
-  DirectFileDownloadProps
-> = props => {
+function DirectDownload(props: DirectFileDownloadProps) {
   const { accessToken } = useSynapseContext()
   const {
     associatedObjectId,
-    entityVersionNumber,
     associatedObjectType,
     fileHandleId,
     displayFileName = false,
     onClickCallback,
     stopPropagation = false,
     iconSvgPropOverrides,
   } = props
-  const { ref, inView } = useInView()
-  const [isExternalFile, setIsExternalFile] = useState<boolean>(false)
-  const [hasFileAccess, setHasFileAccess] = useState<boolean>(false)
-  const [fileEntity, setFileEntity] = useState<FileEntity>()
-  const [externalURL, setExternalURL] = useState<string>()
-  const [fileName, setFileName] = useState<string>('')
-  let mounted = true
 
-  useEffect(() => {
-    if (mounted && inView) {
-      getFileEntityFileHandle()
-    }
-    return () => {
-      mounted = false
-    }
-  }, [accessToken, inView])
+  const fileHandleAssociation = {
+    fileHandleId: fileHandleId,
+    associateObjectId: associatedObjectId,
+    associateObjectType: associatedObjectType,
+  }
+
+  // Get file handle data to ensure we have access / display information
+  const { data } = useGetFileBatch({
+    requestedFiles: [fileHandleAssociation],
+    includeFileHandles: true,
+    // don't get the presigned URL with this query, it may expire before the user clicks the download button
+    includePreSignedURLs: false,
+    includePreviewPreSignedURLs: false,
+  })
```

Comment:
Use react-query to get the file handle information (not the presigned URL, which will still be fetched on click)